widgets/common/src/NodeFilter.tsx

```diff
@@ -133,15 +135,28 @@ export default class NodeFilter extends React.Component {
             .getManager()
             .doGet(fetchUrl, { params })
             .then(data => {
-                let ids = _.chain(data.items || [])
-                    .map(item => item.id)
-                    .uniqWith(_.isEqual)
-                    .value();
+                let allowedOptions = [];
+
                 if (this.isFilteringSetFor(optionsField)) {
-                    ids = _.intersection(ids, this.getAllowedOptionsFor(optionsField));
+                    allowedOptions = this.getAllowedOptionsFor(optionsField);
                 }
 
-                const options = _.map(ids, id => ({ text: id, value: id, key: id }));
+                const options: Stage.Basic.Dropdown.Item.DropdownItemProps[] = Object.entries(
```

Comment:
I turned on TypeScript (removed `@ts-nocheck`) and it turns out this type is invalid :smile: `Dropdown` does not have a registered `Item` property that TypeScript knows about. Thus, we need to make it a bit different.

Also, it would be easier to understand for me if we moved the type annotation to the return type of the last `map` callback. Thus, the final solution that I suggest would be:

```ts
                ).map(([id, displayName]): NonNullable<
                    ComponentProps<typeof Stage.Basic.Dropdown>['options']
                >[number] => ({
```

for the final `map` call (line 154), and then, removing the type annotation from this line:

```suggestion
                const options = Object.entries(
```

Reply:
nice thanks!